File: d1/a.py
import sys

input = open(sys.argv[1]).read().strip()
pairs = input.split("\n")
L1 = []
L2 = []
for p in pairs:
    p = p.split(" ")
    L1.append(int(p[0]))
    L2.append(int(p[3]))

L1.sort()
L2.sort()
i = 0
ans = 0
while i < len(L1):
    ans += abs(L1[i] - L2[i])
    i += 1
print(ans)

File: d1/a.ml
let () = 
    let args = Sys.argv in
    let filename = args.(1) in
    let in_channel = open_in filename in
    let rec parse_input acc1 acc2 = 
        try
            let line = input_line in_channel in
            let list = List.filter ((<>) "") (String.split_on_char ' ' line) in
            let first :: second :: [] =  list in
            parse_input (acc1 @ [int_of_string first]) (acc2 @ [int_of_string second])
        with
        | End_of_file ->
                close_in in_channel;
                acc1, acc2
    in
    let a, b = parse_input [] [] in
    let a = List.sort compare a in
    let b = List.sort compare b in
    let rec result a b acc =
        match a, b with
        | [], [] -> acc
        | h1::t1, h2::t2 -> result t1 t2 (acc + abs (h1 - h2))
        | _ -> failwith "imp"
    in
    print_int (result a b 0)

